pddlgym/parser.py

```diff
@@ -99,6 +99,7 @@ def _create_preconds_pddl_str(self, preconds):
 class PDDLParser:
     """PDDL parsing class.
     """
+    predicates: Dict[str, Predicate]
```

Comment:
what's up with this?

Reply:
I misread `predicates` as being set in one of `PDDLParser`'s methods, I think, and just set the type def at the top of the class def.

Looking back at the code, `predicates` is actually set separately in `PDDLProblemParser` and `PDDLDomainParser`. 

We could

1. set the type hint in the parent `PDDLParser` class for both of them
2.  Set it separately in each
    1. For `PDDLDomainParser`, `predicates` is set in `_parse_domain_predicates`, rather than the init itself. We could either set the type hint in `_parse_domain_predicates`, or the top of the class, like I did here (but for `PDDLDomainParser`.

Reply:
But we can handle that in the split out PR.